extensions/spark/kyuubi-spark-authz/pom.xml

```diff
@@ -331,4 +331,38 @@
         <testOutputDirectory>target/scala-${scala.binary.version}/test-classes</testOutputDirectory>
     </build>
 
+    <profiles>
+        <profile>
+            <id>genpolicy</id>
+            <activation>
+                <property>
+                    <name>ranger.version</name>
+                    <value>2.3.0</value>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.codehaus.mojo</groupId>
+                        <artifactId>build-helper-maven-plugin</artifactId>
```

Comment:
Just out of curiosity, Why do we need this plugin? 

`src/test/gen/scala` should place the golden file instead of the source.

Reply:
The generator class is put in `src/test/gen/scala` rather than in default test source path `src/test/gen/scala` , in order to prevent build and test failed on older Ranger versions.

Reply:
> in order to prevent build and test failed on older Ranger versions.

Golden files should be generated manually by the developer, so the profile should be useless, we can have a ENV, like:
``` shell
KYUUBI_AUTHZ_GENERATE_POLICY_GOLDEN_FILES=1 ./build/mvn scala:run -Pgenpolicy -pl :kyuubi-spark-authz_2.12 -DmainClass=org.apache.kyuubi.plugin.spark.authz.gen.PolicyJsonFileGenerator -Dranger.version=2.3.0
```

1. KYUUBI_AUTHZ_GENERATE_POLICY_GOLDEN_FILES is used to prevent golden files from being generated for each test, so a can be removed plugin `build-helper-maven-plugin`;
2. -Dranger.version=2.3.0 activates the ranger version;


BTW: Does this generator have to depend on Apache Ranger 2.3.0? So, do we need \<activation\>\/\<property\>\/\<ranger.version\> to enable the version condition?

Reply:
The profile `genpolicy` is used for including `PolicyJsonFileGenerator` as additional test source via `build-helper-maven-plugin` plugin.

And here used the `<activation>/<property>/<ranger.version>` to auto activated the `genpolicy` profile to ensure same ranger version used as required in `genpolicy` itself.

For the ENV suggestion, it's not that helpful as runtime switch to control whether generated the policy file as the generator is not runned in tests. Will introduce this switch if we do check the policy file as the same as generator outputs.

Therefore, the command here in method comments is updated as `build/mvn scala:run -pl :kyuubi-spark-authz_2.12 -DmainClass=org.apache.kyuubi.plugin.spark.authz.gen.PolicyJsonFileGenerator`.